pkg/autoscaler/autoscaler_test.go

```diff
@@ -668,3 +628,7 @@ func createEndpoints(ep *corev1.Endpoints) {
 	kubeClient.CoreV1().Endpoints(testNamespace).Create(ep)
 	kubeInformer.Core().V1().Endpoints().Informer().GetIndexer().Add(ep)
 }
+
+func roundedNow() time.Time {
```

Comment:
Is the reason to use `roundedNow` that it prevent flakiness because some stats could be out of scale window if `now()` is used directly?

Reply:
Yes, it basically normalizes the instances of "now" so the test doesn't depend on when exactly it is executed. Especially when adding to "now" in the tests we otherwise risk to jump into other buckets in the calculation. It makes the test deterministic.